Statsman/Core/TimeSplitting/TimeSplittingService.cs

```diff
@@ -0,0 +1,107 @@
+﻿using Statsman.Models;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Watchman.Common.Models;
+using Watchman.DomainModel.Messages;
+
+namespace Statsman.Core.TimeSplitting
+{
+    public class TimeSplittingService
+    {
+        public IEnumerable<TimeStatisticItem> GetStatisticsPerQuarter(IEnumerable<PreGeneratedStatistic> preGeneratedStatistics, IEnumerable<Message> latestMessages, TimeRange expectedTimeRange)
+        {
+            var statisticsPerDay = this.GetStatisticsPerDay(preGeneratedStatistics, latestMessages, expectedTimeRange).ToList();
+            var result = this.SumItems(statisticsPerDay, 90, expectedTimeRange);
+            return result;
+        }
+
+        public IEnumerable<TimeStatisticItem> GetStatisticsPerMonth(IEnumerable<PreGeneratedStatistic> preGeneratedStatistics, IEnumerable<Message> latestMessages, TimeRange expectedTimeRange)
+        {
+            var statisticsPerDay = this.GetStatisticsPerDay(preGeneratedStatistics, latestMessages, expectedTimeRange).ToList();
+            var result = this.SumItems(statisticsPerDay, 30, expectedTimeRange);
+            return result;
+        }
+
+        public IEnumerable<TimeStatisticItem> GetStatisticsPerWeek(IEnumerable<PreGeneratedStatistic> preGeneratedStatistics, IEnumerable<Message> latestMessages, TimeRange expectedTimeRange)
+        {
+            var statisticsPerDay = this.GetStatisticsPerDay(preGeneratedStatistics, latestMessages, expectedTimeRange).ToList();
+            var result = this.SumItems(statisticsPerDay, 7, expectedTimeRange);
+            return result;
+        }
```

Comment:
po co pisać wszędzie var result = ... skoro można od razu zwrócić wynik końcowy bez używania zmiennych o tak ogólnych nazwach.

Reply:
dużo prościej sie później debuguje 
nie trzeba przebudowywać kodu tylko od razu ustawiasz breakpointa i widzisz 

przydatne jak masz fragmenty kodu które prawdopodobnie często trzeba będzie debugować
oczywiście nie wszędzie ma to sens

Reply:
no właśnie tylko do debugowania jest to potrzebne. Myślę, że jeśli ktoś by to debugował to po prostu samemu można na chwilę zrobić taką zmienną do debugowania, a potem wystarczy ją usunąć.

Reply:
no tylko właśnie kod jest dla programisty